### ISSUE TITLE
Generic attribute incompatible with generic property in protocol subclass

### ISSUE THREAD

**Issue by draustin** [microsoft/pyright#6877]
**Describe the bug**
A class with an generic attribute is reported incompatible with a subclass of a protocol with a generic read-only property.

**Code or Screenshots**
```python
from typing import Generic, TypeVar, Protocol

T = TypeVar("T", covariant=True)

class P1(Protocol[T]):
    @property
    def x(self) -> T:
        ...

def f1(_: P1[T]):
    ...

class I(Generic[T]):
    def __init__(self, x: T):
        self.x = x

# OK, because attributes are compatible with read-only properties
# (https://github.com/microsoft/pyright/issues/2371#issuecomment-935116029).
f1(I(1))

# Now extend protocol...
class P2(P1[T], Protocol):
    # Other stuff
    ...

def f2(_: P2[T]):
    ...

f2(I(1))  # False positive.
```

[Pyright playground](https://pyright-play.net/?strict=true&locale=en&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoDiApikSEgMYA0UAKokQGoCGINACuDGBWADYBYAFDDaUALx0GLEAAoARLXk1eAN1ZJmKGONogArkQCUw00Ip9mAZytR2ARlmcw3XnwDatALpGAXMKhAqAABBHAEMngAoIATImAoAA9ZKyI%2BYCMoAFoAPjp-ISCiqAA6MrNhOITgRwB9Xzt7Tx8CorKSivNLGygASVliUnIKZr9owKqoWtrUJBhplLTgGkSG2jHC4sDU9JLEiSTOmtl%2B%2ByNMgGIoAHkAaU6rgDkwAHcoIkSYEhioMJcefjtYQWay2dgAJicTW8HC4AL4GyKV2uMAAFmQoFYYPpgMBxqVyiIiZNgJD6nZwaNWkEgUTSSdZGdMlArgAxZh8VK-MBWOZIVREDpCIA)

Without the generic, it passes ([Pyright playground](https://pyright-play.net/?strict=true&locale=en&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAK4MYAxmADYCwAULbaRQIYDOLhAjABRFgnkUAlAC5aUcVAACCcAgCmeOGIkATOcCgAPLizkVggqAFoAfJhQxRNCTagA6B-To01G4NwD6wziOXiHdk4MzGxQAJJWNq5QHh6oSDCxOnrAADRa3qgwvta24rr6dppQALxaQTQAxFAA8gDS6QBGcqRMAK66UEwwMCBIjW0wcuxMIHJQ5Njd-RTjAO4JABZQY0wqRmAoFHBQMmDyeEjDtNVciz0ILMIA9NdoS22NdpPXEEik4CxgwDDXCHB9NDna5INhtYbXABMAGYAOwcSqgljgyYQOQWIwATmhAFYOBwAGwABkhmMEgRo7i4YS4HEEggq1QAcmA5lA5JohigVLtiGRKAFgqx2ARITwOOlePxKDkbNUajBFgooCwYG1gMA-PZHM5aNFgGKvIRIbKJILdZSxTS6YYoPK6hSgA)).

**VS Code extension or command-line**
VS Code Pylance language server 2023.12.101 (pyright c93d9ebf)

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.345, which I just published. It will also be included in a future release of pylance.
